src/common/hist_util.cc

```diff
@@ -831,7 +831,11 @@ void GHistIndexBlockMatrix::Init(const GHistIndexMatrix& gmat,
  * \brief fill a histogram by zeros in range [begin, end)
  */
 void InitilizeHistByZeroes(GHistRow hist, size_t begin, size_t end) {
+#if defined(XGBOOST_STRICT_R_MODE) && XGBOOST_STRICT_R_MODE == 1
```

Comment:
Could you run a small benchmark on this?  You can  call `std::fill(hist.data() + begin, hist.data() + end, tree::GradStats()` to avoid overhead created by `span`.  If the perf gain by `memset` is negligible, we can mainline it to master.

Reply:
We can use in both cases:
```cpp
using FPType = decltype(tree::GradStats::sum_grad);
FPType* pdata = reinterpret_cast<FPType*>(hist.data());
std::fill(pdata + 2 * begin, pdata + 2 * (begin + end), 0);
```

It should have the same perf as memset

Reply:
Let's come back to this after 1.1.0 release.

Reply:
Tracking issue: #5579